wolfcrypt/src/random.c

```diff
@@ -3415,6 +3415,70 @@ int wc_GenerateSeed(OS_Seed* os, byte* output, word32 sz)
         return 0;
     }
 
+#elif defined(ARDUINO)
+
+    int wc_GenerateSeed(OS_Seed* os, byte* output, word32 sz)
+    {
+        int ret = 0;
+        word32 rand;
+        while (sz > 0) {
+            word32 len = sizeof(rand);
+            if (sz < len)
+                len = sz;
+        /* Get an Arduino framework random number */
+        #if defined(__arm__)
+            /* See: https://github.com/avrxml/asf/tree/master/sam/utils/cmsis/sam3x/include */
+            #if defined(__SAM3A4C__)
+                #ifndef TRNG
+                    #define TRNG (0x400BC000U)
+                #endif
+            #elif defined(__SAM3A8C__)
+                #ifndef TRNG
+                    #define TRNG (0x400BC000U)
+                #endif
+            #elif defined(__SAM3X4C__)
+                #ifndef TRNG
+                    #define TRNG (0x400BC000U)
+                #endif
+            #elif defined(__SAM3X4E__)
+                #ifndef TRNG
+                    #define TRNG (0x400BC000U)
+                #endif
+            #elif defined(__SAM3X8C__)
+                #ifndef TRNG
+                    #define TRNG (0x400BC000U)
+                #endif
+            #elif defined(__SAM3X8E__)
+                /* This is the Arduino Due */
+                #ifndef TRNG
+                    #define TRNG (0x400BC000U)
+                #endif
+            #elif  defined(__SAM3A8H__)
+                #ifndef TRNG
+                    #define TRNG (0x400BC000U)
+                #endif
+            #else
+                #ifndef TRNG
+                    #error "Unknown TRNG for this device"
+                #endif
+            #endif
+
+            srand(analogRead(0));
+            rand = trng_read_output_data(TRNG);
+        #elif defined(__STM32__)
+            /* TODO: confirm this is proper random number on Arduino STM32 */
+            rand = random();
```

Comment:
We have STM32 code in random.c. Most of them have a TRNG.

Reply:
Do you think it is best to use the existing STM32 detection for Arduino, or still pull out the Arduino separately as is done in this PR and deal with the STM32 Arduino specifics there?

Reply:
Depends if the stm32xxxx_hal.h HAL is available. If so then just use the STM32_RNG section. Arduino is not specific HW, just an IDE/SDK platform.

Reply:
As this PR is only for Espressif ESP32 and Arduino (only Due tested)... and the fact that I cannot get my STM32 board to even recognize any sort of networking at this time, I'd like to put off STM32 work for a future PR. Here's why:

Additional targets are installed via the Arduino [Boards Manager](https://docs.arduino.cc/software/ide-v2/tutorials/ide-v2-board-manager/) and the files end up (in my case for Windows) in:

```
C:\Users\%USERNAME%\AppData\Local\Arduino15\packages\
```

I have 19 different ` stm32*hal.h` files, for instance `stm32l4xx_hal.h` in 

```
 C:\Users\%USERNAME%\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.7.1\system\Drivers\STM32L4xx_HAL_Driver\Inc
```

I did try to use the [STM32_RNG](https://github.com/wolfSSL/wolfssl/blob/f9bf96d9bac779cdf495e583e145b4de23b393d5/wolfcrypt/src/random.c#L2885) in my Arduino `user_settings.h`, located here:

```
C:\Users\%USERNAME%\Documents\Arduino\libraries\wolfSSL\src\user_settings.h
```

I assumed that it would be a simple matter of determining the exact chip and including the appropriate file to resolve error such as this:

```
C:\workspace\wolfssl-arduino\IDE\ARDUINO\sketches\wolfssl_server_stm32\random.c: In function 'wc_GenerateSeed':
C:\workspace\wolfssl-arduino\IDE\ARDUINO\sketches\wolfssl_server_stm32\random.c:3017:32: error: 'RCC_AHB2Periph_RNG' undeclared (first use in this function)
 3017 |         RCC_AHB2PeriphClockCmd(RCC_AHB2Periph_RNG, ENABLE);
      |                                ^~~~~~~~~~~~~~~~~~
```

Quite surprisingly, the `RCC_AHB2Periph_RNG` is found nowhere in the over 7,200 files in that directory. :/